src/server/zset_family.cc

```diff
@@ -1436,6 +1532,57 @@ void ZSetFamily::ZScan(CmdArgList args, ConnectionContext* cntx) {
   }
 }
 
+void ZSetFamily::ZUnion(CmdArgList args, ConnectionContext* cntx) {
+  OpResult<UnionArgs> union_args_res = ParseUnionArgs(args);
```

Comment:
In _theory_, ZUnion and ZUnionStore could be handled by one function, that just decides in the end whether to store the result or to reply. This would need some knowledge of the transaction framework